src/core/include/units/magnitude.h

```diff
@@ -225,7 +325,23 @@ static constexpr bool is_base_power_pack_valid = all_base_powers_valid<BPs...> &
  */
 template<BasePower auto... BPs>
   requires (detail::is_base_power_pack_valid<BPs...>)
-struct magnitude {};
+struct magnitude {
+  // Whether this magnitude represents an integer.
+  static constexpr bool is_magnitude_integral = (detail::is_integral(BPs) && ...);
+  friend constexpr bool is_integral(magnitude) { return is_magnitude_integral; }
+
+  // Whether this magnitude represents a rational number.
+  static constexpr bool is_magnitude_rational = (detail::is_rational(BPs) && ...);
+  friend constexpr bool is_rational(magnitude) { return is_magnitude_rational; }
+
+  // The value of this magnitude, expressed in a given type.
+  template<typename T>
+    requires (
+        std::is_floating_point_v<T>
+        || (std::is_integral_v<T> && (detail::is_integral(BPs) && ...)))
+  static constexpr T value = detail::checked_static_cast<T>(
+      (detail::compute_base_power<T>(BPs) * ...));
```

Comment:
I am a bit afraid that this will require the usage of a `template` disambiguation for dependent names in our code. This is also why `std::get()` is not a member function of a `std::tuple` or `std::variant`. Maybe it would better as a `to_value()` non-member function?